### ISSUE TITLE
SSH Server and client crashes

### ISSUE THREAD

**Issue by Mido22** [mscdex/ssh2#571]
I played around a bit with setting `ident` of ssh server:

- with no value set, no issues, displays default server software name/version identifier.
- with any string set, displays that string to client as software version.
- with '' (empty) string, displays the default value
- with ' ' (single space) string, it works fine with default ssh client, displays empty sting as remote software version but when ssh2 node module is used as client, both the server and the client crashes.

##  Outputs for `ident` set as ' '

with default linux ssh client:

    $ ssh -p 2201  foo@localhost -v
    OpenSSH_6.6.1, OpenSSL 1.0.1f 6 Jan 2014
    debug1: Reading configuration data /etc/ssh/ssh_config
    debug1: /etc/ssh/ssh_config line 19: Applying options for *
    debug1: Connecting to localhost [127.0.0.1] port 2201.
    debug1: Connection established.
    debug1: identity file /home/mido/.ssh/id_rsa type 1
    debug1: identity file /home/mido/.ssh/id_rsa-cert type -1
    debug1: identity file /home/mido/.ssh/id_dsa type -1
    debug1: identity file /home/mido/.ssh/id_dsa-cert type -1
    debug1: identity file /home/mido/.ssh/id_ecdsa type -1
    debug1: identity file /home/mido/.ssh/id_ecdsa-cert type -1
    debug1: identity file /home/mido/.ssh/id_ed25519 type -1
    debug1: identity file /home/mido/.ssh/id_ed25519-cert type -1
    debug1: Enabling compatibility mode for protocol 2.0
    debug1: Local version string SSH-2.0-OpenSSH_6.6.1p1 Ubuntu-2ubuntu2.8
    debug1: Remote protocol version 2.0, remote software version  
    debug1: no match:  
    debug1: SSH2_MSG_KEXINIT sent
    debug1: SSH2_MSG_KEXINIT received
    debug1: kex: server->client aes128-ctr hmac-sha1 none
    debug1: kex: client->server aes128-ctr hmac-sha1 none
    debug1: sending SSH2_MSG_KEX_ECDH_INIT
    debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
    ...
    
But when connected with `ssh2` node module as client:

    $ node client.js 
    events.js:182
          throw er; // Unhandled 'error' event
          ^

    Error: Handshake failed: signature verification failed
        at onKEXDH_REPLY (/mnt/genji/middleEarth/ssh-test/node_modules/ssh2-streams/lib/ssh.js:2903:15)
        at SSH2Stream.<anonymous> (/mnt/genji/middleEarth/ssh-test/node_modules/ssh2-streams/lib/ssh.js:215:45)
        at emitOne (events.js:115:13)
        at SSH2Stream.emit (events.js:210:7)
        at parse_KEXDH_REPLY (/mnt/genji/middleEarth/ssh-test/node_modules/ssh2-streams/lib/ssh.js:4209:8)
        at parse_KEX (/mnt/genji/middleEarth/ssh-test/node_modules/ssh2-streams/lib/ssh.js:4171:14)
        at parsePacket (/mnt/genji/middleEarth/ssh-test/node_modules/ssh2-streams/lib/ssh.js:4006:12)
        at SSH2Stream._transform (/mnt/genji/middleEarth/ssh-test/node_modules/ssh2-streams/lib/ssh.js:667:13)
        at SSH2Stream.Transform._read (_stream_transform.js:190:10)
        at SSH2Stream._read (/mnt/genji/middleEarth/ssh-test/node_modules/ssh2-streams/lib/ssh.js:251:15)

And server crash output:

    $ node server.js 
    Client connected!
    events.js:182
          throw er; // Unhandled 'error' event
          ^

    Error: KEY_EXCHANGE_FAILED
        at onDISCONNECT (/mnt/genji/middleEarth/ssh-test/node_modules/ssh2-streams/lib/ssh.js:2197:15)
        at SSH2Stream.<anonymous> (/mnt/genji/middleEarth/ssh-test/node_modules/ssh2-streams/lib/ssh.js:203:5)
        at emitMany (events.js:146:13)
        at SSH2Stream.emit (events.js:223:7)
        at parsePacket (/mnt/genji/middleEarth/ssh-test/node_modules/ssh2-streams/lib/ssh.js:3743:10)
        at SSH2Stream._transform (/mnt/genji/middleEarth/ssh-test/node_modules/ssh2-streams/lib/ssh.js:667:13)
        at SSH2Stream.Transform._read (_stream_transform.js:190:10)
        at SSH2Stream._read (/mnt/genji/middleEarth/ssh-test/node_modules/ssh2-streams/lib/ssh.js:251:15)
        at SSH2Stream.Transform._write (_stream_transform.js:178:12)
        at doWrite (_stream_writable.js:371:12)

Client code:
```javascript
var Client = require('ssh2').Client
 
var conn = new Client()
conn.on('ready', () => {
  console.log('Client :: ready')
  conn.end();
}).connect({
  host: 'localhost',
  port: 2201,
  username: 'foo',
  password: 'bar'
})
```

Server code:
```javascript
var ssh2 = require('ssh2')
var utils = ssh2.utils
var fs = require('fs')
var pubKey = utils.genPublicKey(utils.parseKey(fs.readFileSync('testSSH.pub')))
 
new ssh2.Server({
  ident: ' ',
  hostKeys: [fs.readFileSync('testSSH')]
}, function(client) {
  console.log('Client connected!'); 
  client
    .on('end', () =>  console.log('Client disconnected'))
    .on('authentication', ctx => ctx.accept())
    .on('ready', function() {
      console.log('Client authenticated!');
      client.on('session', accept => accept())
    })
}).listen(2201, '127.0.0.1')
```

**Comment by Mido22** [comment]
Update, after a bit of testing, noticed that `ident` does not have to be just a single space, **any string** with a space in the end leads to this crash. Not sure if the right solution is for user should have a listener to the 'error' event or just trim the ident before using.

**Comment by Mido22** [comment]
created a [linked issue](https://github.com/mscdex/ssh2-streams/issues/72) in ssh-streams

**Comment by mscdex** [comment]
Closing this as this should be fixed now in mscdex/ssh2-streams@09c5884
